### PR TITLE
chore: enable extension tests in ci

### PR DESCRIPTION
**Pull request by Askir**
I replaced the ENABLE_ flags for each API provider with simply checking if a key is present. I think this makes more sense so you don't have to provide both env vars?

I also added a download fixture to the ollama tests so that the necessary models are downloaded during the test runs. This is quite slow, but I guess we can add caching later.

Other than that this was quite straight forward. Only issue is that voyageai already hits rate limits with my free key now.

### CODE REVIEW

File: projects/extension/build.py [hunk 1]
```diff
@@ -648,7 +648,7 @@ def docker_build() -> None:
 def docker_run() -> None:
     cmd = " ".join(
         [
-            "docker run -d --name pgai-ext -p 127.0.0.1:5432:5432 -e POSTGRES_HOST_AUTH_METHOD=trust",
+            "docker run -d --name pgai-ext --network host -p 127.0.0.1:5432:5432 -e POSTGRES_HOST_AUTH_METHOD=trust",
```
Comment (JamesGuthrie) [inline projects/extension/build.py:651]: Why was this change necessary?

File: projects/extension/tests/test_ollama.py [hunk 2]
```diff
@@ -1,16 +1,58 @@
...
+    raise TimeoutError(f"Timeout waiting for model {model} to be ready")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def ensure_models(ollama_host):
+    """
+    Ensure required models are downloaded before running tests.
+    This fixture runs automatically at the start of the test session.
+    """
+    required_models = ["llama3", "llava:7b"]
```
Comment (JamesGuthrie) [inline projects/extension/tests/test_ollama.py:50]: Daaamn, these models are large (each ~4.7 GB).

File: projects/extension/build.py (hunk 1 above)
Comment (Askir) [reply projects/extension/build.py:651]: because the ollama container and the container in which we run tests, otherwise cannot talk to each other.
The alternative would be to run both in a docker-compose or otherwise share a network. But I felt like this was simpler and should also help for local development if you run ollama separately.

File: projects/extension/tests/test_ollama.py (hunk 2 above)
Comment (JamesGuthrie) [reply projects/extension/tests/test_ollama.py:50]: I think for testing we're mostly trying to assert that it does _something_. I would suggest that we switch `llama3` to `llama3.2:1b`, and `llava:7b` to `smollm:138m`.

Comment (Askir) [reply projects/extension/tests/test_ollama.py]: Updated and 5 minutes now, seems good to me.

**Review by JamesGuthrie — APPROVED** [review]
I added a commit with something that I need to get tests to pass locally. Otherwise LGTM.

**Comment by jgpruitt** [comment]
nice!
